commands/team_e2e_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package commands
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
+)
+
+func (s *MmctlE2ETestSuite) TestDeleteTeamsCmdF() {
+	s.SetupTestHelper().InitBasic()
+
+	s.RunForAllClients("Error deleting team which does not exist", func(c client.Client) {
+		printer.Clean()
+		teamName := "existingName"
+		cmd := &cobra.Command{}
+		args := []string{""}
+		args[0] = teamName
+		cmd.Flags().String("display_name", "newDisplayName", "Team Display Name")
+		cmd.Flags().Bool("confirm", true, "")
+
+		_ = deleteTeamsCmdF(c, cmd, args)
+		s.Len(printer.GetErrorLines(), 1)
+		s.Require().Equal("Unable to find team '"+teamName+"'", printer.GetErrorLines()[0])
+	})
+
+	s.Run("Permission error while deleting a valid team", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+		args := []string{""}
+		args[0] = s.th.BasicTeam.Name
+		cmd.Flags().String("display_name", "newDisplayName", "Team Display Name")
+		cmd.Flags().Bool("confirm", true, "")
+
+		_ = deleteTeamsCmdF(s.th.Client, cmd, args)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 1)
+		s.Require().Equal("Unable to delete team '"+s.th.BasicTeam.Name+"' error: : You do not have the appropriate permissions., ", printer.GetErrorLines()[0])
+	})
+
+	s.Run("Delete a valid team", func() {
```

Comment:
I think we can run this using `RunForSystemAdminAndLocal`.

Reply:
I wasnt able to get it working for SystemAdminClient. This is error I get when I run with SystemAdminClient.
Unable to delete team 'faketeamaxkfw6' error: : Permanent team deletion feature is not enabled. Please contact your System Administrator.

Reply:
Interesting, this is probably cause of [EnableAPITeamDeletion](https://docs.mattermost.com/administration/config-settings.html#enable-api-team-deletion) not being enabled on the server, which is the default.
Maybe we could add a test to check for this specific error when running as a System Admin.

Reply:
Added.

Reply:
I think we could use `RunForSystemAdminAndLocal` in this test previously enabling API team deletion, and then explicitly disable it on the next test case and test how `s.th.SystemAdminClient` can't remove the team but `s.th.LocalClient` can.

I'm thinking on something like:

```go
s.RunForSystemAdminAndLocal("With API team deletion enabled, both sysadmin and local should be able to delete a team", func(c client.Client) {
    // create team
    // update config to enable API deletion
    // delete team and ensure it's gone
})

s.Run("With API team deletion disabled, only local client is able to delete a team", func() {
    // create team
    // update config to disable API deletion
    // run delete command with sysadmin and check that team is still there
    // run same command with local and check that the team is gone
})
```

Reply:
Added.